### PR TITLE
HW 18. Playwright actions, assertions

### PR DESCRIPTION
**Pull request by viaylya1**

### CODE REVIEW

File: tests/pw_actions_assertions/registration_form.spec.js [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+    const signupData = {
+      name: 'Vika',
+      lastName: 'Bryzhak',
+      email: 'aqa-vika@test.com',
+      password: 'Test1Test',
+      repeatPassword: 'Test1Test'
+    };
+
+    const signInBtn = page.locator('button', { hasText: 'Sign In' });
+    await signInBtn.click();
```
Comment (sttaran) [inline tests/pw_actions_assertions/registration_form.spec.js]: we have sign up button on welcome page

File: tests/pw_actions_assertions/registration_form.spec.js [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+        LongInvalidValue: 'Invalid And LongNameWithMoreThanTwentyCharacters'
+
+      };
+
+      await test.step('Empty "Name" value should give an error', async () => {
+        await nameField.focus();
+        await nameField.blur();
+        await expect(invalidField).toHaveText('Name required');
+        await expect(signupPopup).toHaveScreenshot('Name required.png');
+        await expect(registerBtn).toBeDisabled();
```
Comment (sttaran) [inline tests/pw_actions_assertions/registration_form.spec.js:80]: add check for input border color if invalid value is entered

File: tests/pw_actions_assertions/registration_form.spec.js [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+        await expect(invalidField).toHaveText('Name is invalid');
+        await nameField.clear();
+
+        await nameField.fill(testNameData.invalidValue4);
+        await expect(invalidField).toHaveText('Name is invalid');
+        await nameField.clear();
+
+        await nameField.fill(testNameData.notEnglishValue);
+        await expect(invalidField).toHaveText('Name is invalid');
+        await nameField.clear();
```
Comment (sttaran) [inline tests/pw_actions_assertions/registration_form.spec.js]: use loop when you have multiple variants, it is really good moment to use DDT

**Review by sttaran — CHANGES_REQUESTED** [review]

File: tests/pw_actions_assertions/registration_form.spec.js (hunk 2 above)
Comment (viaylya1) [reply tests/pw_actions_assertions/registration_form.spec.js:80]: I check it at the end of negative test for each field. 
image


Should I check it after each negative test.step?

File: tests/pw_actions_assertions/registration_form.spec.js (hunk 1 above)
Comment (viaylya1) [reply tests/pw_actions_assertions/registration_form.spec.js]: Fixed

File: tests/pw_actions_assertions/registration_form.spec.js (hunk 2 above)
Comment (viaylya1) [reply tests/pw_actions_assertions/registration_form.spec.js:80]: I checked, border is checked in all fields

File: tests/pw_actions_assertions/registration_form.spec.js (hunk 3 above)
Comment (viaylya1) [reply tests/pw_actions_assertions/registration_form.spec.js]: All tests are re-written with using "for of" loop.

**Review by sttaran — APPROVED** [review]

**Pull request merged**